opslevel/resource_opslevel_group.go

```diff
@@ -0,0 +1,186 @@
+				Description: "List of users' email who are members of the group.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"teams": {
+				Type:        schema.TypeList,
+				Description: "The teams where this group is the direct parent. Accepts ID or Alias",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceGroupCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	members := []opslevel.MemberInput{}
+	for _, member := range getStringArray(d, "members") {
+		members = append(members, opslevel.MemberInput{Email: member})
+	}
+
+	teams := []opslevel.IdentifierInput{}
+	for _, team := range getStringArray(d, "teams") {
+		teams = append(teams, *opslevel.NewIdentifier(team))
+	}
+
+	input := opslevel.GroupInput{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Members:     &members,
+		Teams:       &teams,
+	}
+	if parent, ok := d.GetOk("parent"); ok {
+		input.Parent = opslevel.NewIdentifier(parent.(string))
+	}
+	resource, err := client.CreateGroup(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(resource.Id.(string))
+
+	return resourceGroupRead(d, client)
+}
+
+func resourceGroupRead(d *schema.ResourceData, client *opslevel.Client) error {
+	id := d.Id()
+
+	resource, err := client.GetGroup(id)
+	if err != nil {
+		return err
+	}
+
+	groupMembers, err := resource.Members(client)
+	if err != nil {
+		return err
+	}
+	members := flattenMembersArray(groupMembers)
+
+	childTeams, err := resource.ChildTeams(client)
+	if err != nil {
+		return err
+	}
+	teams := flattenTeamsArray(childTeams)
+
+	if err := d.Set("alias", resource.Alias); err != nil {
+		return err
+	}
+	if err := d.Set("name", resource.Name); err != nil {
+		return err
+	}
+	if err := d.Set("description", resource.Description); err != nil {
+		return err
+	}
+	if err := d.Set("parent", resource.Parent.Id); err != nil {
+		return err
+	}
+	if err := d.Set("members", members); err != nil {
+		return err
+	}
+	if err := d.Set("teams", teams); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceGroupUpdate(d *schema.ResourceData, client *opslevel.Client) error {
+	input := opslevel.GroupInput{}
+
+	members := []opslevel.MemberInput{}
+	for _, member := range getStringArray(d, "members") {
+		members = append(members, opslevel.MemberInput{Email: member})
+	}
+
+	teams := []opslevel.IdentifierInput{}
+	for _, team := range getStringArray(d, "teams") {
+		teams = append(teams, *opslevel.NewIdentifier(team))
+	}
+
+	if d.HasChange("name") {
+		input.Name = d.Get("name").(string)
+	}
+	if d.HasChange("description") {
+		input.Description = d.Get("description").(string)
+	}
+	if d.HasChange("parent") {
+		if parent, ok := d.GetOk("parent"); ok {
+			input.Parent = opslevel.NewIdentifier(parent.(string))
+		} else {
+			input.Parent = nil
```

Comment:
Not sure how this is supposed to work... when I try to apply with the parent field commented out, it doesn't remove the parent even though the plan says that it will.  Tried this on the CLI and I was unable to clear out the parent so not sure if I'm missing how to do that or if it's a bug.

Reply:
Resolving this since it was an issue on the opslevel-go side